### PR TITLE
docs: Adding documentation for visualization tool

### PR DESCRIPTION
**Pull request by yahia3200**
## Description

Adding the technical documentation for visualization tool and the work done during GSoC23 project

### CODE REVIEW

File: visualization_tool/docs/DOC.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+We can divide the architecture of the application into three parts:
+
+- Types: The types definitions for objects used in the application.
+- Parsing: The logic for parsing and extracting the data from the GCP Scanner.
+- UI: The code and components hierarchy for displaying the data in the UI.
+
+### Types Definition
+
+The types layer contains the types definitions that used across the app. It works as a middle layer between the parsing layer and the UI layer Where it defines what data shape is expected from the parsing logic.
```
[the remark this answers is not in the document]
Comment (mshudrak) [reply visualization_tool/docs/DOC.md:31]: the UI layer Where/the UI layer where

File: visualization_tool/docs/DOC.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+There are two main types of data that is extracted from the GCP Scanner:
```
Comment (mshudrak) [inline visualization_tool/docs/DOC.md:33]: that is/that are

File: visualization_tool/docs/DOC.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+  - `projectId`: The ID of the project that the IAM policy belongs to.
+  - `file`: The file that the IAM policy was found in.
+
+The `types/resources.ts` file also contains types related to the structure of the GCP scanner output file which are:
+
+- `OutputFile`: The GCP scanner output file object schema.
+
+### Parsing
+
+The logic for parsing and extracting the data from the GCP Scanner output files is defined in the `components/ControlMenu/Controller.ts` file. and it contains two main functions:
```
Comment (mshudrak) [inline visualization_tool/docs/DOC.md:70]: . and/ and

File: visualization_tool/docs/DOC.md [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+```
+for each key in the project keys:
+  if the key is a resource:
+    parse the resources in the array and add them to the resources array
+```
+
+To extract the resources from the GCP scanner output file, We are looping through each key in the project keys. If the key is a resource, we are parsing the resource and adding it to the resources array.
+
+To check if the key is a resource, we are checking it is in the `availableResourceTypes` array that is defined in the `types/resources.ts` file. The `availableResourceTypes` array contains the types of resources that we want to extract from the GCP scanner output file. These are the types of resources that are well tested and supported by the visualizer.
```
Comment (mshudrak) [inline visualization_tool/docs/DOC.md:87]: we are checking it is/ we are checking if it is

File: visualization_tool/docs/DOC.md [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+Currently, there are two views (pages) in the app:
+
+- `Resources Page`: This view displays the resources as cards in a grid. Each card contains the resource name, type, and status. The user can see more details about the resource by clicking on the details button on the card. The user can also filter and sort the resources based on their needs.
+
+- `IAM Policies Page`: This view displays the IAM policies in a table. The policies are grouped by the project ID. The user can see all members of a policy by clicking on the expand button in the policy row. The user can also filter the IAM policies based on their needs.
+
+## Deployment
+
+The visualizer is deployed as a static website that ships with the GCP scanner package and is served using the `http.server` module in Python. The `src/gui/app.py` python script is responsible for serving the visualizer static files that should be in the `src/gui/static` folder.
```
Comment (mshudrak) [inline visualization_tool/docs/DOC.md:120]: the visualizer/ the visualizer's

File: visualization_tool/docs/DOC.md [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+
+### Integration with GCP Scanner
+
+The `npm run build` command in configured in the `package.json` file to build the visualizer and output the static files in the `src/gui/static` folder and the `MANIFEST.in` file is configured to include the static files in the `src/gui/static` folder in the GCP scanner package.
```
Comment (mshudrak) [inline visualization_tool/docs/DOC.md:124]: in/is

Comment (mshudrak) [reply visualization_tool/docs/DOC.md:124]: folder and the/folder. The

File: visualization_tool/docs/DOC.md [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+
+We also modified the `python-publish.yml` GitHub action workflow to build the visualizer and include the static files in the GCP scanner package when publishing to PyPI.
```
Comment (mshudrak) [inline visualization_tool/docs/DOC.md:126]: in/to

**Review by mshudrak — CHANGES_REQUESTED** [review]
I left a couple of minor comments. PTAL.

**Comment by yahia3200** [comment]
Done fixing all mentioned typos

**Comment by yahia3200** [comment]
Do you have any other comments on the document and the information in it? and how much information did you get from it?

**Comment by yahia3200** [comment]
Do you have any other comments on the document and the information in it? and how much information did you get from it?

**Comment by mshudrak** [comment]
All good, thanks, merging.

**Review by mshudrak — APPROVED** [review]

**Pull request merged**